guvnor-ala/guvnor-ala-registry-vfs/src/main/java/org/guvnor/ala/registry/vfs/VFSRegistryHelper.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.ala.registry.vfs;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.guvnor.ala.marshalling.Marshaller;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.uberfire.io.IOService;
+import org.uberfire.java.nio.IOException;
+import org.uberfire.java.nio.file.DirectoryStream;
+import org.uberfire.java.nio.file.FileSystem;
+import org.uberfire.java.nio.file.Path;
+
+/**
+ * Helper class for implementing the different VFS based registries that are backed by using VFSRegistryEntry elements.
+ */
+@ApplicationScoped
+public class VFSRegistryHelper {
+
+    protected static final String PROVISIONING_BRANCH = "master";
+
+    protected static final String PROVISIONING_PATH = "provisioning";
+
+    private static final Logger logger = LoggerFactory.getLogger(VFSRegistryHelper.class);
+
+    private VFSMarshallerRegistry marshallerRegistry;
+
+    private IOService ioService;
+
+    private FileSystem fileSystem;
+
+    private Path provisioningRootPath;
+
+    private VFSRegistryEntryMarshaller entryMarshaller;
+
+    public VFSRegistryHelper() {
+        //Empty constructor for Weld proxying
+    }
+
+    @Inject
+    public VFSRegistryHelper(final VFSMarshallerRegistry marshallerRegistry,
+                             final @Named("ioStrategy") IOService ioService,
+                             final @Named("systemFS") FileSystem fileSystem) {
+        this.marshallerRegistry = marshallerRegistry;
+        this.ioService = ioService;
+        this.fileSystem = fileSystem;
+    }
+
+    @PostConstruct
+    protected void init() {
+        try {
+            provisioningRootPath = fileSystem.getPath(PROVISIONING_BRANCH,
+                                                      PROVISIONING_PATH);
+            logger.debug("provisioningRootPath: " + provisioningRootPath.toUri());
+        } catch (Exception e) {
+            //uncommon error
+            logger.error("An error was produced during VFS registries directory initialization.",
```

Comment:
Related to the comments above, the init() case will "never" happen, the readEntries will hopefully occur with low probabilty, but is different since e.g. if the conent of a registry entry was manually modified (I don't know who could be interested on doing this) we can have an unmarshalling error, or whatever. And this is why this case is also different to the init().